connectors/services/sync.py

```diff
@@ -132,18 +108,11 @@ async def _one_sync(self, connector, es, sync_now):
             await connector.close()
 
     async def run(self):
-        if "PERF8" in os.environ:
-            import perf8
-
-            async with perf8.measure():
-                return await self._run()
-        else:
-            return await self._run()
-
-    async def _run(self):
         """Main event loop."""
 
-        self.connectors = BYOIndex(self.config["elasticsearch"])
+        await super().run()
```

Comment:
I'm not sure about this pattern of usage of base implementation.

What do you think instead in wrapping this function the following way:

```
class BaseService:
  def start(self):
    # do logic
    # then run the inherited class's method that does the logic
    await self.run()

  def run(self):
    raise NotImpelementedException # should be overriden in the class that inherits BaseService

```

Reply:
Dunno though, maybe it's a python-default way of doing so, but I find it more error-prone:

- You can forget calling `await.super().run()` in child classes
- You have to put this code in specific place in child classes for it to wor
- Once base class's `run` function grows and does more complex logic, it might want to wrap child class's `run` method from both sides - before and after execution

Reply:
made the change in https://github.com/elastic/connectors-python/pull/337/commits/b41fcb72af21f6dfab7c7b0d2f5a602e95e0d29f